src/azul/chalice.py

```diff
@@ -468,12 +471,34 @@ def catalog(self) -> str:
     def _controller(self, controller_cls: Type[C], **kwargs) -> C:
         return controller_cls(app=self, **kwargs)
 
+    def content_security_policy(self, nonce: Optional[str] = None) -> str:
+        def q(s: str) -> str:
+            return f"'{s}'"
+
+        def s(*args: str | None) -> str:
+            return ' '.join(filter(None, args))
+
+        self = q('self')
```

Comment:
To avoid overriding the reference to the `self` parameter.
```suggestion
        self_ = q('self')
```